crates/router/src/core/errors/error_handlers.rs

```diff
@@ -1,24 +1,27 @@
-use actix_web::{dev::ServiceResponse, middleware::ErrorHandlerResponse, ResponseError};
+use actix_web::{body, dev::ServiceResponse, middleware::ErrorHandlerResponse, ResponseError};
 use http::StatusCode;
 
 use super::ApiErrorResponse;
-pub fn custom_error_handlers<B>(
+pub fn custom_error_handlers<B: body::MessageBody + 'static>(
     res: ServiceResponse<B>,
 ) -> actix_web::Result<ErrorHandlerResponse<B>> {
     let error_response = match res.status() {
         StatusCode::NOT_FOUND => ApiErrorResponse::InvalidRequestUrl,
         StatusCode::METHOD_NOT_ALLOWED => ApiErrorResponse::InvalidHttpMethod,
         _ => ApiErrorResponse::InternalServerError,
     };
-    let (req, _) = res.into_parts();
-    let res = error_response.error_response();
+    let (req, res) = res.into_parts();
+    let res = match res.error() {
```

Comment:
Add logs for this change - Old and New one.

Reply:
> Old response with error
> <img width="1380" alt="Screenshot 2023-02-15 at 5 57 56 PM" src="https://user-images.githubusercontent.com/51093026/219027373-e68c321c-3d11-4e54-897b-6a87298742b0.png">

> New response with error
> <img width="1315" alt="Screenshot 2023-02-15 at 6 00 26 PM" src="https://user-images.githubusercontent.com/51093026/219027887-93ab96d6-a9cf-42f8-9571-8a5b1cb1e19d.png">